src/TS_HW5/hw5.ts

```diff
@@ -0,0 +1,127 @@
+/*  Створіть класи Circle, Rectangle, Square і Triangle.
+    У кожного з них є загальнодоступний метод calculateArea.
+    У кожної фігури є загальнодоступні властивості - колір і назва, які не можна змінювати після створення.
+    У Square і Rectangle зі свого боку є ще додатковий метод print, який виводить рядок із формулою розрахунку площі.
+*/
+enum Colors {
+  Red = 'Red',
+  Orange = 'Orange',
+  Yellow = 'Yellow',
+  Green = 'Green',
+  LigthBlue = 'Ligth Blue',
+  Blue = 'Blue',
+  Violet = 'Violet',
+}
+
+enum Names {
+  Circle = 'Circle',
+  Rectangle = 'Rectangle',
+  Square = 'Square',
+  Triangle = 'Triangle',
+}
+
+abstract class Figure {
+  protected abstract readonly Color: Colors;
+  protected abstract readonly Name: Names;
+  protected abstract CalculateArea(numberA: number, numberB?: number, numberC?: number): number;
```

Comment:
All works until a new requirement with a new figure is introduced, such as a polygon. It would be beneficial for you to modify the abstract class according to SOLID principles. Although we haven't talked about it, it might be helpful for you to become familiar with the following points. The current method of adding dependencies as function parameters in the Abstract class is not the best way to do it. It would be better to create a property for each figure that needs area calculation.

Reply:
Alternatively, you can consider implementing a rest operator for function arguments.

Reply:
Added private fields and set/get methods for properties